src/Illuminate/Auth/composer.json

```diff
@@ -15,10 +15,10 @@
     ],
     "require": {
         "php": "^7.2",
-        "illuminate/contracts": "5.9.*",
-        "illuminate/http": "5.9.*",
-        "illuminate/queue": "5.9.*",
-        "illuminate/support": "5.9.*"
+        "illuminate/contracts": "^6.0",
```

Comment:
I'm not sure what's Laravel strategy with `contracts` on semver, but wouldn't it be safer to limit to patch release.

Here an example:

* I want to use `illuminate/auth=~6.0.0` which limit `>=6.0.0` to `<6.1.0`.
* Laravel decided to add a new contract (Say `GuardContract::foo()` method) to `illuminate/contracts` for `Auth` in `6.1.0` (incremental non-breaking changes) and finally release `v6.1.0`.
* On my project, `illuminate/contracts` version `v6.1.0` is permitted based on `^6.0` constraint and now the code break because `GuardContract` missing `foo()` method.

Reply:
Unless we always enforce that `illuminate/contracts` can only be modified between major release and all the `minor` and `patch` release is simply a re-tag to `v6.0.0` commit.

Of course trivial commit such as comments and docblock should be okay on non-major release.

Reply:
You cannot add new interface methods since that would break compatibility. This isn't really a problem in this version requirement, but how any framework code is developed. New interface, changing classes to abstract, or many other things that require implementors to change something, will break backward compatibility. 

A really nice list of actions that is allowed or not allowed is available at Symfony's homepage. https://symfony.com/doc/current/contributing/code/bc.html

Reply:
> Laravel decided to add a new contract (Say GuardContract::foo() method) to illuminate/contracts for Auth in 6.1.0 (incremental non-breaking changes) and finally release v6.1.0.

This will never happen as it's a BC break. New methods will only be added in major releases.